scrypto/src/resource/resource_flags.rs

```diff
@@ -1,23 +1,37 @@
+const RESOURCE_FLAGS_SHIFT: u8 = 0;
+const RESOURCE_FLAGS_MASK: u64 = 0xffff_ffff;
+
+macro_rules! resource_flags {
+    ( $f:expr ) => {
+        ($f as u64) << RESOURCE_FLAGS_SHIFT
+    }
+}
+
+#[inline]
+pub fn resource_flags_are_valid(flags: u64) -> bool {
+    (flags & RESOURCE_FLAGS_MASK) == flags
```

Comment:
This flag check seems wrong to me. Should be `flags & !RESOURCE_FLAGS_MASK == 0`.

Reply:
Never mind. It's the same.